sqlmodel/sql/sqltypes.py

```diff
@@ -11,6 +11,8 @@ class AutoString(types.TypeDecorator):  # type: ignore
 
     def load_dialect_impl(self, dialect: Dialect) -> "types.TypeEngine[Any]":
         impl = cast(types.String, self.impl)
-        if impl.length is None and dialect.name == "mysql":
+        if impl.length is None and (
+            dialect.name == "mysql" or dialect.name == "mariadb"
+        ):
```

Comment:
```suggestion
        if impl.length is None and dialect.name in ("mysql", "mariadb"):
```
A bit more compact and easier to read